charts/microgateway/README.md.gotmpl

```diff
@@ -1,153 +1,365 @@
+
+
+### Advanced DSL app configuration
+
+In case that the [Simple DSL configuration](#simple-dsl-configuration) does not suite, the advanced configuration options might help. The following use cases might require this kind of configuration:
+
+
+Use Case 1)
+* Virtual Host 1: abc.com -> Mapping 1: / -> Backend Service 1
+* Virtual Host 2: xyz.com -> Mapping 2: / -> Backend Service 1
+
+Use Case 2)
+* Virtual Host 1: abc.com -> Mapping 1: /      -> Backend Service 1
+                          -> Mapping 2: /auth/ -> Backend Service 1
+
+Use Case 3)
+* Virtual Host 1: abc.com -> Mapping 1: / -> Backend Service 1
+* Virtual Host 2: xyz.com -> Mapping 2: / -> Backend Service 2
+
+These use cases are only examples which could also occur slightly different. But all of them have in common that they have more than one Virtual Hosts, Mappings or Backend Services. When ever this is the case, the [Advanced DSL app configuration](#advanced-dsl-app-configuration) should be preferred over the [Simple DSL configuration](#simple-dsl-configuration).
+
+
+**Example:**
+
 custom-values.yaml
 ```
 config:
-  expert:
-    apache: |
-      LogLevel debug
-  apps: | 
+  IPHeader:
+    trustedProxies:
+      - 10.0.0.0/28
+  apps:
     - virtual_host:
-        hostname: custom-hostname
-      backend:
-        hostname: backend-service
+        hostname: virtinc.com
       mappings:
-        - name: app
+        - name: webapp
           entry_path: /
-          backend_path: /app/
+          operational_mode: integartion
+          session_handling: enforce_session
+        - name: api
+          entry_path: /api/
+          session_handling: ignore_session
+          openapi:
+            spec_file: /config/virtinc_api_openapi.json
+      backend:
+        hostname: custom-backend
+
+redis:
+  enabled: true
 ```
 
-### Custom DSL
-In case that both other configuration options are not sufficient, create a custom config using `config.dsl`. All the configuration of the DSL can be used.
-Overwrites all config defaults of this chart. 
+**Parameters which can be used**:
+* `config.apps` - **must** be used.
+* `config.*`
+* Only the following `config.default.*` parameters:
+  * `config.default.virtualHost.tls.*`
+  * `config.default.backend..tls.*`
 
-## Configuration
+**Parameters which cannot be used**:
+* `config.dsl` - This would switch to the [Expert DSL app configuration](#expert-dsl-app-configuration)
+* `config.default.mapping.*`
+* `config.default.backend.protocol`
+* `config.default.backend.hostname`
+* `config.default.backend.port`
 
-{{ template "chart.valuesTable" . }}
+### Expert DSL configuration
 
-## Chart dependencies
-The helm chart has optional dependencies which can be activated.
+In case that the [Advanced DSL app configuration](#advanced-dsl-app-configuration) does not suite, the expert configuration options might help. The following reasons might this kind of configuration:
```

Comment:
the expert configuration options might help.
There is no other option then using the expert DSL if the Advanced DSL doesn't suite, 

the expert configuration options  must be used.

Reply:
I changed this part slightly. Please have a look again.

Reply:
Okay looks good.